<launch>
    <arg name="use_cuda" default="true" />
    <arg name="show" default="false" />
    <arg name="mapping_type" default="tsdf" />
    <arg name="texture_enabled" default="false" />
    <arg name="enable_mesher" default="false" />
    <arg name="output_map" default="false" />
    <arg name="max_disp_particles" default="10000000" />
    <arg name="max_mesh" default="300000" />
    <arg name="max_ray_length" default="3.1" />
    <arg name="min_ray_length" default="0.3" />
    <arg name="enable_submap" default="true" />
    
    <node pkg="taichislam" type="taichislam_node.py" name="taichislam_node" output="screen" >
        <param name="use_cuda" value="$(arg use_cuda)" type="boolean" />
        <param name="enable_rendering" value="$(arg show)" type="boolean" />
        <param name="mapping_type" value="$(arg mapping_type)" type="string" />
        <param name="texture_enabled" value="$(arg texture_enabled)" type="boolean" />
        <param name="enable_mesher" value="$(arg enable_mesher)" type="boolean" />
        <param name="enable_submap" value="$(arg enable_submap)" type="boolean" />
        <param name="output_map" value="$(arg output_map)" type="boolean" />
        
        <param name="max_ray_length" value="$(arg max_ray_length)" type="double" />
        <param name="min_ray_length" value="$(arg min_ray_length)" type="double" />
    
        <param name="disp/max_disp_particles" value="$(arg max_disp_particles)" type="int" />
        <param name="disp/max_mesh" value="$(arg max_mesh)" type="int" />

        <remap from="~pointcloud" to="/depth_estimation/point_cloud" />
        <!-- <remap from="~pose" to="/vins_estimator/camera_pose" /> -->
        <remap from="~pose" to="/d2vins/camera_pose_1" />
        <remap from="~traj" to="/d2pgo/pgo_traj" />
        <remap from="~frame_local" to="/d2vins/frame_local" />

        <rosparam>
            disp_ceiling: 1.0
            disp_floor: -1.5
            texture_compressed: true
            voxel_size: 0.1
            disp:
                res_x: 1920
                res_y: 1080
        </rosparam>
    </node>
</launch>